scripts/push_docs_to_branch.sh

```diff
@@ -1,11 +1,22 @@
 #!/bin/bash
 
+# In case this script fails in our CI workflows, you can run it locally.
+# For instance, if the doc version for 2.2.10 fails to be pushed, you
+# can do:
+#
+# TMP_DOCS_FOLDER=/tmp/documentation DOCS_BRANCH=documentation GITHUB_REF=refs/tags/2.2.10 ./scripts/push_docs_to_branch.sh
+#
+# This command will push the changes to the `documentation` branch of this repository.
+#
+# Please make sure to have the proper dependencies installed with `make install install-docs`
+# and that the docs are pre-built with `make prepare-docs`.
+
 set -Eeuo pipefail
 
 TODAY=`date "+%Y%m%d"`
 # we build new versions only for minors and majors
 PATTERN_FOR_NEW_VERSION="^refs/tags/[0-9]+\\.[0-9]+\\.0$"
-PATTERN_FOR_PATCH_VERSION="^refs/tags/[0-9]+\\.[0-9]+\\.[1-9]+$"
+PATTERN_FOR_MICRO_VERSION="^refs/tags/[0-9]+\\.[0-9]+\\.[1-9]+[0-9]*$"
```

Comment:
Seems a bit clearer to me ("no 0 in the beginning, then all numbers are allowed")
Should we also check the "no 0 in the beginning" for majors / minors?
```suggestion
PATTERN_FOR_MICRO_VERSION="^refs/tags/[0-9]+\\.[0-9]+\\.[1-9][0-9]*$"
```

Reply:
yeah that's true, we do not need the `+` 👍🏻 . And yes, I think it's more accurate even though it doesn't change the result.

Reply:
actually no, for minors and majors it's accurate. We want to be able to match `0.0.12`, `2.0.22`, etc...